pkg/installmanager/installmanager_test.go

```diff
@@ -145,6 +145,11 @@ func TestInstallManager(t *testing.T) {
 			expectPasswordSecret:          true,
 			expectProvisionMetadataUpdate: true,
 		},
+		{
+			name:        "infraID already set on cluster provision", // fatal error
+			existing:    []runtime.Object{testClusterDeployment(), testClusterProvisionWithInfraIDSet()},
+			expectError: true,
```

Comment:
It would be nice if we could verify that this is in fact the error we expected. But that's a latent issue, something for the backlog.